.changeset/olive-worms-run.md

```diff
@@ -0,0 +1,5 @@
+---
+'skuba': patch
+---
+
+template/koa-rest-api: Use AsyncLocalStorage to track logger context
```

Comment:
```suggestion
template/koa-rest-api: Use [AsyncLocalStorage](https://nodejs.org/docs/latest-v16.x/api/async_context.html#asynchronous-context-tracking) to track logger context
```

Reply:
> This looks excellent and it sounds like there is little performance penalty with Node.js 16: [nodejs/node#34493 (comment)](https://github.com/nodejs/node/issues/34493#issuecomment-845094849). I think our Lambdas could similarly benefit from the same approach.
> 
> It's pretty trivial but I wonder if we should upstream something to https://github.com/seek-oss/koala to enforce this pattern.

Ooooo that might be an idea. We could potentially initialise it via seek koala and ship the middleware from there too